base/logging.go

```diff
@@ -34,6 +34,9 @@ const (
 
 type Level int32
 
+// DeferredLog is a log function that can be logged at a later date.
+type DeferredLog func()
```

Comment:
What about changing this to `DeferredLogFn` to highlight that it's a function rather than a log message/statement?   On my first reading of `var deferredLogs []DeferredLog` I had no idea that was a slice of functions.